### PR TITLE
Feature: APP-1639  - Cached Proposal status

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- dynamically calculate cached proposal status

Task: [APP-1639](https://aragonassociation.atlassian.net/browse/APP-1639)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1639]: https://aragonassociation.atlassian.net/browse/APP-1639?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/web-app/src/utils/proposals.ts [hunk 1]
```diff
@@ -1072,3 +1084,46 @@ export function augmentProposalWithCachedExecution(
     return {...proposal, status: ProposalStatus.EXECUTED};
   }
 }
+
+/**
+ * Calculate a proposal's status
+ * @param proposal Proposal
+ * @returns status for proposal
+ */
+function calculateProposalStatus(proposal: DetailedProposal): ProposalStatus {
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/proposals.ts:1093]: How will we call the next wrapper? `figureOutProposalStatus`? :D

File: packages/web-app/src/context/apolloClient.tsx [hunk 2]
```diff
@@ -233,8 +234,12 @@ const pendingMultisigExecutionVar = makeVar<PendingMultisigExecution>(
 // iffy about this structure
 export type CachedProposal = Omit<
   DetailedProposal,
-  'creationBlockNumber' | 'executionBlockNumber' | 'executionDate'
->;
+  'creationBlockNumber' | 'executionBlockNumber' | 'executionDate' | 'status'
+> & {
```
Comment (WissenIstNacht) [inline packages/web-app/src/context/apolloClient.tsx:238]: Oh well. We tried. haha

**Review by WissenIstNacht — APPROVED** [review]
Thanks man. This is another piece of code that really wasn't designed to but keeps on getting more complex because with every addition it turns out we need another one.

Kudos for managing.

File: packages/web-app/src/context/apolloClient.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/context/apolloClient.tsx:238]: I know, I know! 😭

File: packages/web-app/src/utils/proposals.ts (hunk 1 above)
Comment (Fabricevladimir) [reply packages/web-app/src/utils/proposals.ts:1093]: Hahaha yes most certainly!

**Pull request merged**